llvm/test/CodeGen/SPIRV/hlsl-intrinsics/firstbithigh.ll

```diff
@@ -0,0 +1,37 @@
+; RUN: llc -verify-machineinstrs -O0 -mtriple=spirv-unknown-unknown %s -o - | FileCheck %s
+; RUN: %if spirv-tools %{ llc -O0 -mtriple=spirv-unknown-unknown %s -o - -filetype=obj | spirv-val %}
```

Comment:
Do you have spirv-tools installed?

There is an error in DXCs SPIRV backend
`error: firstbithigh is currently limited to 32-bit width components when targetting SPIR-V`

That makes me think there might be a problem with non 32 bit width values that you use in this test. short of running the shader I don't know how to confirm. Maybe someone with more knowledge could weigh in.

Reply:
I removed the non-32 bit tests, except error tests, since it only supports 32 bit integers.

Reply:
I dunno if you have to remove the tests i think the return type of those tests were the problem.

Reply:
If you look at FindSMsb and FindUMsb in https://registry.khronos.org/SPIR-V/specs/unified1/GLSL.std.450.html, it says:

> This instruction is currently limited to 32-bit width components.

For DXC, we opted to just not implement the 64-bit version. We can choose to do something different in Clang. See https://github.com/microsoft/DirectXShaderCompiler/issues/4702 for that discussion.

This is an issue for 16-bit values as well.